core/forms.py

```diff
@@ -43,61 +37,14 @@ def __init__(self, event_choices=None, *args, **kwargs):
         if event_choices:
             self.fields['event'].queryset = event_choices
 
-    def invite_to_slack(self, email, name):
-        try:
-            user_invite(email, name)
-        except (ConnectionError, SlackerError) as e:
-            self._errors.append(
-                'Slack invite unsuccessful, reason: {}'.format(e))
-
-    def notify_existing_user(self, user):
-        content = render_to_string('emails/existing_user.html', {
-            'user': user,
-            'event': self.cleaned_data['event']
-        })
-        subject = 'You have been granted access to new Django Girls event'
-        self.send_email(content, subject, user)
-
-    def notify_new_user(self, user):
-        content = render_to_string('emails/new_user.html', {
-            'user': user,
-            'event': self.cleaned_data['event'],
-            'password': self._password,
-            'errors': self._errors,
-        })
-        subject = 'Access to Django Girls website'
-        self.send_email(content, subject, user)
-
-    def send_email(self, content, subject, user):
-        msg = EmailMessage(subject,
-                           content,
-                           "Django Girls <hello@djangogirls.org>",
-                           [user.email])
-        msg.content_subtype = "html"
-        msg.send()
-
     def save(self, *args, **kwargs):
         assert self.is_valid()
         self._errors = []
         email = self.cleaned_data['email']
         event = self.cleaned_data['event']
-
-        user, created = User.objects.get_or_create(email=email)
-        event.team.add(user)
-        if created:
-            self._password = User.objects.make_random_password()
-            user.first_name = self.cleaned_data['name'].split(' ')[0]
-            user.last_name = self.cleaned_data[
-                'name'].replace(user.first_name, '')
-            user.is_staff = True
-            user.is_active = True
-            user.set_password(self._password)
-            user.save()
-            user.groups.add(1)
-            self.invite_to_slack(email, user.first_name)
-            self.notify_new_user(user)
-        else:
-            self.notify_existing_user(user)
+        first_name = self.cleaned_data['name'].split(' ')[0]
+        last_name = self.cleaned_data['name'].replace(first_name, '')
```

Comment:
This will give strange results for a name like `"Jo Johnson"`. Something like `first_name, _, last_name = self.cleaned_data['name'].partition(' ')` should work better and handle edge cases too (no space present in the name, more than one space present, ...)